src/server/main_service.cc

```diff
@@ -1448,7 +1509,7 @@ void Service::RegisterCommands() {
              &EvalValidator)
       << CI{"EVALSHA", CO::NOSCRIPT | CO::VARIADIC_KEYS, -3, 3, 3, 1}.MFUNC(EvalSha).SetValidator(
              &EvalValidator)
-      << CI{"EXEC", kExecMask, 1, 0, 0, 0}.MFUNC(Exec)
+      << CI{"EXEC", CO::LOADING | CO::NOSCRIPT, 1, 0, 0, 1}.MFUNC(Exec)
```

Comment:
First, this is no longer GLOBAL by default

Second, the step is now 1. This is because some parts of tx rely on it for reading arguments. We virtually "append" arguments to EXEC to schedule it ahead for example

Reply:
if we lock ahead

We read the keys from shard args based on their 'step'. If we schedule ahead with EXEC, then we store the keys inside the per-shard args

Reply:
I would like to add a comment about it here or where you do it